pkg/joinserver/grpc_deviceregistry.go

```diff
@@ -23,11 +23,27 @@ import (
 	"go.thethings.network/lorawan-stack/pkg/crypto/cryptoservices"
 	"go.thethings.network/lorawan-stack/pkg/crypto/cryptoutil"
 	"go.thethings.network/lorawan-stack/pkg/errors"
+	"go.thethings.network/lorawan-stack/pkg/events"
 	"go.thethings.network/lorawan-stack/pkg/joinserver/provisioning"
 	"go.thethings.network/lorawan-stack/pkg/ttnpb"
 	"go.thethings.network/lorawan-stack/pkg/types"
 )
 
+var (
+	evtCreateEndDevice = events.Define(
+		"js.end_device.create", "create end device",
+		ttnpb.RIGHT_APPLICATION_DEVICES_READ,
```

Comment:
Also `ttnpb.RIGHT_APPLICATION_DEVICES_WRITE`?

Reply:
No, because write-only doesn't give you read access to devices created by others